fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubqueryToApply.java

```diff
@@ -434,8 +434,14 @@ private Pair<LogicalPlan, Optional<Expression>> addApply(SubqueryExpr subquery,
                                 Preconditions.checkState(logicalProject.getOutputs().size() == 1,
                                         "Scalar subuqery's should only output 1 column");
                                 Slot aggSlot = agg.toSlot();
-                                replaceMap.put(aggSlot, new Alias(new Nvl(aggSlot,
-                                        notNullableAggFunc.resultForEmptyInput())));
+                                // we should keep the exprId, so that the parent plan can bind this slot
+                                replaceMap.put(aggSlot,
+                                        new Alias(
+                                                aggSlot.getExprId(),
```

Comment:
maybe we need give the agg function a new exprid and replace it in the subquery, to avoid generate same exprid for two different slot in one plan.

Reply:
I think this `generate same exprid for two different slot in one plan` is not a problem, ensure different exprId maybe high cost, because it maybe exists under multiple level children, for example

in this case, you should found all `#382` to replace
```
LogicalProject(projects=[count(*)#382])
                  |
           LogicalApply
                  |
    LogicalProject(projects=[ifnull(count(*)#382, 0) AS `ifnull(count(*), 0)`#693])
                 |
           LogicalApply
                |
       LogicalAggregate(groupByExpr=[], outputExpr=[count(*) AS `count(*)`#382])
                |
              ....
```

Reply:
line 385: change 
```
if (!ctx.subqueryIsAnalyzed(subqueryExpr)) {
    tmpPlan = addApply(subqueryExpr, tmpPlan.first,
        subqueryToMarkJoinSlot, ctx, conjunct,
        isProject, subqueryExprs.size() == 1, isMarkJoinSlotNotNull);
}
```
to
```
if (!ctx.subqueryIsAnalyzed(subqueryExpr)) {
    tmpPlan = addApply(subqueryExpr, tmpPlan.first,
        subqueryToMarkJoinSlot, ctx, tmpPlan.second,
        isProject, subqueryExprs.size() == 1, isMarkJoinSlotNotNull);
}
```
this should work and no need to keep expr id unchanged